pyproject.toml

```diff
@@ -17,9 +17,9 @@ python = ">=3.8,<4.0"
 abnf = "1.1.1"
 web3 = "^6.0.0"
 python-dateutil = "2.8.2"
-eth-account = "^0.8.0"
-eth-utils = "^2.2.0"
-eth-typing = "^3.4.0"
+eth-account = ">=0.8.0,<0.11"
```

Comment:
I'm in favour of simply upgrading to 0.10, while keeping the semver requirement

Reply:
Upgrading to 0.10 would work for me.  But like you pointed out, 0.10 should be valid, no?  

> Because siwe (2.4.0) depends on eth-account (>=0.8.0,<0.9.0)
>  and no versions of siwe match >2.4.0,<3.0.0, siwe (>=2.4.0,<3.0.0) requires eth-account (>=0.8.0,<0.9.0).

This is the conflict that brought me here.  In my deps, I'm using `eth-account = "^0.10"`.  And `poetry lock` apparently can't resolve this.  Any idea?

Reply:
Unpacking the wheel, I see this in the dist-info `METADATA` file:

> Requires-Dist: eth-account (>=0.8.0,<0.9.0)

Weird

Reply:
Also, if I try using git source for my poetry deps to bypass the wheel:

```toml
siwe = { git = "https://github.com/spruceid/siwe-py.git", tag = "v2.4.0" }
```

I still get a similar version conflict.

> Because chaosnet depends on siwe (2.4.0) @ git+https://github.com/spruceid/siwe-py.git@v2.4.0 which depends on eth-account (^0.8.0), eth-account is required.
> So, because thispackage depends on eth-account (^0.10), version solving failed.

Does the caret behave differently for `0.` as a major version?

Reply:
Yeah, here's a clarifying issue in Poetry's GH issue tracker:

> You can imagine the ^ as saying: allow any version up to the next breaking change according to [SemVer](https://semver.org/). Because SemVer allows breaking changes in a minor version bump for versions <1.0, a ^0.7.0 is equivalent to >=0.7.0,<0.8.0.

https://github.com/python-poetry/poetry/issues/5370#issuecomment-1081027876

That's unexpected.  So, I'll revert my other changes for eth-typing and eth-utils.  Though I think the range is necessary if we want to allow for a wider range of versions for eth-account.